airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/StatePersistence.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.config.persistence;
+
+import static io.airbyte.db.instance.configs.jooq.generated.Tables.STATE;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.StateType;
+import io.airbyte.config.StateWrapper;
+import io.airbyte.db.Database;
+import io.airbyte.db.ExceptionWrappingDatabase;
+import io.airbyte.protocol.models.AirbyteGlobalState;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.AirbyteStreamState;
+import io.airbyte.protocol.models.StreamDescriptor;
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.jooq.DSLContext;
+import org.jooq.JSONB;
+import org.jooq.Record;
+import org.jooq.RecordMapper;
+import org.jooq.impl.DSL;
+
+public class StatePersistence {
+  private final ExceptionWrappingDatabase database;
+
+  public StatePersistence(final Database database) {
+    this.database = new ExceptionWrappingDatabase(database);
+  }
+
+  /**
+   * Get the current State of a Connection
+   *
+   * @param connectionId
+   * @return
+   * @throws IOException
+   */
+  public Optional<StateWrapper> getCurrentState(final UUID connectionId) throws IOException {
+    final List<StateRecord> records = this.database.query(ctx -> getStateRecords(ctx, connectionId));
+
+    if (records.isEmpty())
```

Comment:
nit: Does it respect the code style? I think that we need `{}`.

Reply:
Added. Weirdly the style check didn't complain.